### PR TITLE
Update to work with htsget

### LINKED ISSUE #47: Update images to work with htsget-rs
As an gdi-dev
I want to have the latest images from the sensitive data archive, including the re-encrypt service
In order to be able to work with the htsget-rs new version

Tasks:
- [ ] Update the images in the storage and interfaces repo
- [ ] Update the image or htsget-rs in the htsget repo
- [ ] Make sure they work together, by requesting an encrypted file via htsget

### PR DESCRIPTION
**Pull request by MalinAhlberg**
This PR closes #47 .

Include changes from [feature/test-htsget-reencrypt](https://github.com/GenomicDataInfrastructure/starter-kit-storage-and-interfaces/tree/featute/test-htsget-reencrypt).
The image of sda-download used in this setup is built from https://github.com/neicnordic/sensitive-data-archive/pull/820.
To be used with [GenomicDataInfrastructure/starter-kit-htsget/feature/rust-htsget](https://github.com/GenomicDataInfrastructure/starter-kit-htsget/tree/feature/rust-htsget).

**Testing**
Make sure that all containers can start and that data can be loaded.
Test to run together with  [GenomicDataInfrastructure/starter-kit-htsget/feature/rust-htsget](https://github.com/GenomicDataInfrastructure/starter-kit-htsget/tree/feature/rust-htsget) (as specified in that PR).

### CODE REVIEW

File: docker-compose-demo.yml [hunk 1]
```diff
@@ -102,7 +102,28 @@ services:
...
+    command: [sda-reencrypt]
+    container_name: reencrypt
+    depends_on:
+      credentials:
+        condition: service_completed_successfully
+    ports:
+      - "50443:50443"
+    restart: always
+    networks:
+      - public
```
Comment (jbygdell) [inline docker-compose-demo.yml]:
```suggestion
```
This shouldn't be needed

File: docker-compose-demo.yml [hunk 2]
```diff
@@ -102,7 +102,28 @@ services:
...
+
+  reencrypt:
+    image: ghcr.io/neicnordic/sensitive-data-archive:v0.3.25
+    command: [sda-reencrypt]
+    container_name: reencrypt
+    depends_on:
+      credentials:
+        condition: service_completed_successfully
+    ports:
+      - "50443:50443"
```
Comment (jbygdell) [inline docker-compose-demo.yml]:
```suggestion
```
This shouldn't be needed

**Review by jbygdell — APPROVED** [review]
Some things that technically aren't needed.

File: docker-compose-demo.yml (hunk 1 above)
Comment (nanjiangshu) [reply docker-compose-demo.yml]: That's true.

File: docker-compose-demo.yml (hunk 2 above)
Comment (nanjiangshu) [reply docker-compose-demo.yml]: Right, I mapped the port for testing the reencrypt service directly in the CLI and it's not needed when `download` is in the same network as `reencrypt`.

**Review by nanjiangshu — APPROVED** [review]

File: docker-compose-demo.yml (hunk 2 above)
Comment (MalinAhlberg) [reply docker-compose-demo.yml]: Thanks! Fixed in c20bfe0

File: docker-compose-demo.yml (hunk 1 above)
Comment (MalinAhlberg) [reply docker-compose-demo.yml]: Fixed in c20bfe0.

**Pull request merged**